cityofaustin/atd-moped
# Use Data Grid in the Projects List

## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/15353

This PR migrates the project list table from Material Table to [DataGrid](https://mui.com/x/react-data-grid/)

It does not satisfy all of the requirements from the issue linked above, namely it does not allow users to pin columns. That is a feature available only in the paid version of Data Grid. 

It also does not have consistent row height, but instead I let it have auto row height depending on the row's content. 

## Testing
**URL to test:** 
https://deploy-preview-1275--atd-moped-main.netlify.app/moped/projects

**Steps to test:**

I want to come up with a better test guide than "do everything you can on the table", but that's really what has to be done to test this. 

I am testing viewing every column, toggling column visibility off and on. Sorting by various columns, advanced filters and normal filters. 

Questions I have:
1. Do you like how the columns are ordered? We have been for the most part tacking on new columns at the end, should "Created by" for example be at the end of the table? https://mui.com/x/react-data-grid/column-ordering/ The pro version offers column ordering
2. We can search and advanced search by the project description, but as far as I can remember, we have never showed it in the table -- should we add it? Or have we showed it in the table but decided to remove it? 
3. DataGrid's column visibility model follows a columnName: isVisible pattern, instead of the columnName: isHidden pattern material table does. I changed the logic, but didnt rename the config to be DEFAULT_VISIBLE_COLUMNS, is everyone okay with that? 
4. While testing I noticed that the parent project name sorting isnt working -- its sorting on parent project id instead of the name. I temporarily turned this off, its also broken in prod. I was going to ask if you think we should write a custom sort function for this, but I think I have a way around it. 

---
#### Ship list
